arangod/Aql/ConstFetcher.cpp

```diff
@@ -0,0 +1,85 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Christoph Uhde
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Aql/ConstFetcher.h"
+
+#include "Aql/AqlItemBlock.h"
+#include "Aql/BlockFetcher.h"
+#include "Aql/FilterExecutor.h"
+#include "ConstFetcher.h"
+
+
+using namespace arangodb;
+using namespace arangodb::aql;
+
+ConstFetcher::ConstFetcher()
+    :  _currentRow{CreateInvalidInputRowHint{}} {}
+
+void ConstFetcher::injectBlock(std::shared_ptr<InputAqlItemBlockShell> block){
+    _currentBlock = std::move(block);
+    _rowIndex = 0;
+    // get number input regs
+}
+
+std::pair<ExecutionState, InputAqlItemRow> ConstFetcher::fetchRow() {
+  if (_currentBlock == nullptr || !indexIsValid()) {
```

Comment:
`_currentBlock == nullptr` is checked in `indexIsValid()`
I am good with only the later to be present here.